### ISSUE TITLE
Is this relevant with Django 4.0?

### ISSUE THREAD

**Issue by AqeelAT** [jazzband/django-redis#585]
Since Django 4.0 has native redis support, do we still need to use this library?

**Comment by WisdomPill** [comment]
**DISCLAIMER** I am biased, since I am a user and a maintainer of `django-redis`.

Yes, but the native support of `django` 4 is very basic, 
if you need a compressor, serializer, connection pool or sentinel support you can not do that with `django` native support, 
in some cases it would be very difficult (sentinel and compressor), 
in other not so much (serializers).

Furthermore, if you need more control over how the keys are stored, 
like expire_at and millisecond precision you could only do it with `django-redis`. 
Locks are also something nice that in my day job makes a difference, 
for syncing celery tasks for example, 
`django-redis` can help you create distributed redis locks while `django` native support could not.
Another thing, for performance, is that `django` does not supports `hiredis` while `django-redis` does.
Also, in this case I do not really know, I would be glad to be corrected, 
`django` natively does not support options for the connection pool.
For certificates also I `django` does not support them.

`django-redis` is highly configurable, you can easily define different clients, or use sharded client for that manner.

Have a look [here](https://github.com/django/django/pull/14437#issuecomment-848613586)

But, if you need very basic caching, 
then using `django-redis` or `django` native support should not make any difference.
Maybe in the future `django` will do the same things as `django-redis`.

In short, `django-redis` is more redis specific, while `django` native support is more basic.

**Comment by saranglakare** [comment]
I feel you should integrate with `django`. There's no point in having extra package for such a core functionality. At the same time, I agree with you that `django-redis` is for those who have fixated on using `redis` for their cache.

**Comment by FlipperPA** [comment]
It would be awesome to see some of these features make it upstream into core. Not having to wait for packages to add support for new versions of Django is a big win.

**Comment by getup8** [comment]
FWIW it does look like Django supports `hiredis` and potentially connection pools?

From lower down the page at https://docs.djangoproject.com/en/4.0/topics/cache/#redis-1 :

> Here’s an example configuration for a redis based backend that selects database 10 (by default Redis ships with 16 logical databases), specifies a [parser class](https://github.com/redis/redis-py#parsers) (redis.connection.HiredisParser will be used by default if the hiredis-py package is installed), and sets a custom [connection pool class](https://github.com/redis/redis-py#connection-pools) (redis.ConnectionPool is used by default)

**Comment by alexandernst** [comment]
We're at Django 5.x now, how much of this comparison / reasons to use django-redis are still valid?

**Comment by WisdomPill** [comment]
now it's even more diverged than before, because we support sentinel and more commands for different data structures such as hashmaps and sets. Future developments are going to add more support for specifics of redis, as with memcached as well, if you want or need a more advanced use case, you have to go outside of plain basic django
